### ISSUE TITLE
Expand support for "executable" accessory files

### ISSUE THREAD

**Issue by hinerm** [imagej/imagej-updater#109]
Currently, `executable` files are treated specially and directly downloaded: they bypass the `update` subdirectory when installing, and an `.old` copy is created to signify deletion.

With [Jaunch](https://github.com/apposed/jaunch) we now have necessary `toml` accessory files that are required for the launcher to work. This means they should be treated the same way by the launcher: bypassing the `update` directory and with a `.old` copy.

Basically, if we're checking for `executable` we should probably check if it's a `toml` file, as well.

[This](https://github.com/imagej/imagej-updater/blob/7465c0e1193a2aed67788d54837706d5596e677e/src/main/java/net/imagej/updater/Installer.java#L133-L147) is one key location, but there are almost certainly others that need updating as well.

**Comment by hinerm** [comment]
Note that a temporary hack can be done on the update site side, by manually adding `executable = "true"` to the `toml` entries in the `db.xml.gz`

**Comment by hinerm** [comment]
note that the `Props.class` (and `.java`?) file(s) are also needed

**Comment by ctrueden** [comment]
Thanks @hinerm. About `Props.java`: it is not needed.

**Comment by hinerm** [comment]
Ah there's just a warning about `Props` being missing. I didn't realize that I was not able to upload it even with my changes

**Comment by ctrueden** [comment]
On POSIX-friendly systems: is the executable bit of the file used to discern whether the file should be marked "executable" in the DB? And on Windows: does it use the .exe extension to decide?

And conversely: when installing an "executable" file from an update site, does the Updater set the executable bit on those files on POSIX-friendly systems? If so, that means the TOML files will be marked +x, which is suboptimal. Tolerable as a hack, yeah, but it would be nice not to conflate the two related-but-distinct ideas of "executable" and "part of the launcher machinery".
